keylime/src/tpm.rs

```diff
@@ -1929,18 +1993,251 @@ pub mod testing {
+            //HashingAlgorithm::Sha3_384, // Not supported by swtpm
+            //HashingAlgorithm::Sha3_512, // Not supported by swtpm
+            //HashingAlgorithm::Sha1, // Not supported by swtpm
+        ];
+        let sign_algs = [
+            SignAlgorithm::RsaSsa,
+            SignAlgorithm::RsaPss,
+            // - ECC keys creation requires this: https://github.com/parallaxsecond/rust-tss-esapi/pull/464
+            //   Probably this will be released on tss_esapi version 8.0.0, which includes API
+            //   breakage
+            // SignAlgorithm::EcDsa,
+            // SignAlgorithm::EcSchnorr,
+        ];
+
+        for sign in sign_algs {
+            for hash in hash_algs {
+                let r = ctx.create_ak(ek_handle, hash, sign);
+                assert!(r.is_ok());
+
+                let ak = r.unwrap(); //#[allow_ci]
+
+                let r = ctx.load_ak(ek_handle, &ak);
+                assert!(r.is_ok());
+            }
+        }
+    }
+
+    #[test]
+    fn test_create_idevid() {
+        let asym_algs = [AsymmetricAlgorithm::Rsa, AsymmetricAlgorithm::Ecc];
+        let hash_algs = [
+            HashingAlgorithm::Sha256,
+            HashingAlgorithm::Sha384,
+            //HashingAlgorithm::Sha512, // Not supported by swtpm
+            //HashingAlgorithm::Sm3_256, // Not supported by swtpm
+            //HashingAlgorithm::Sha3_384, // Not supported by swtpm
+            //HashingAlgorithm::Sha3_512, // Not supported by swtpm
+            //HashingAlgorithm::Sha1, // Not supported by swtpm
+        ];
+
+        let mut ctx = Context::new().unwrap(); //#[allow_ci]
+
+        for asym in asym_algs {
+            for hash in hash_algs {
+                let r = ctx.create_idevid(asym, hash);
+                assert!(r.is_ok());
+            }
+        }
+    }
+
+    #[test]
+    fn test_create_iak() {
+        let mut ctx = Context::new().unwrap(); //#[allow_ci]
+
+        let asym_algs = [AsymmetricAlgorithm::Rsa, AsymmetricAlgorithm::Ecc];
+        let hash_algs = [
+            HashingAlgorithm::Sha256,
+            HashingAlgorithm::Sha384,
+            //HashingAlgorithm::Sha512, // Not supported by swtpm
+            //HashingAlgorithm::Sm3_256, // Not supported by swtpm
+            //HashingAlgorithm::Sha3_384, // Not supported by swtpm
+            //HashingAlgorithm::Sha3_512, // Not supported by swtpm
+            //HashingAlgorithm::Sha1, // Not supported by swtpm
+        ];
+
+        for asym in asym_algs {
+            for hash in hash_algs {
+                let r = ctx.create_iak(asym, hash);
+                assert!(r.is_ok())
+            }
+        }
+    }
+
+    #[test]
+    fn test_activate_credential() {
+        let mut ctx = Context::new().unwrap(); //#[allow_ci]
+        let r = ctx.create_ek(EncryptionAlgorithm::Rsa, None);
+        assert!(r.is_ok());
+
+        // Create EK
+        let ek_result = r.unwrap(); //#[allow_ci]
```

Comment:
This is a matter of taste but I would say it's easier to read if you write:
```suggestion
        let ek_result = ctx.create_ek(EncryptionAlgorithm::Rsa, None)
            .expect("unable to create EK");
```

Reply:
Thanks, will follow the suggestion!